### PR TITLE
Fix misleading tracer cache invalidation comment and simplify logic

### PR DESCRIPTION
**Pull request by torosent**
# Summary

## What changed?
- Removed the unused `_cachedOtel` variable from `trace-helper.ts`
- Simplified the tracer cache check from `if (!_cachedTracer || _cachedOtel !== otel)` to `if (!_cachedTracer)`
- Replaced the misleading cache invalidation comment with an accurate explanation of why caching is safe

## Why is this change needed?
The previous comment claimed "The cache is invalidated if the global tracer provider changes (e.g., in tests)" but the code only compared the OTEL API module reference (`_cachedOtel !== otel`). Since `getOtelApi()` always returns the same module object (due to Node's `require()` cache and internal `_otelApiLoaded` guard), this condition was never true after the first call — making `_cachedOtel` dead code and the comment misleading.

The tracer caching is actually safe because the OTEL JS SDK's `trace.getTracer()` returns a proxy tracer that dynamically delegates to the current global tracer provider. Provider swaps (e.g., via `setGlobalTracerProvider()` in tests) are handled transparently without needing to recreate the tracer. The new comment explains this correctly.

## Issues / work items
- N/A

---

# Project checklist
- [x] Release notes are not required for the next release
- [x] Backport is not required
- [x] All required tests have been added/updated (unit tests, E2E tests)
  - No new tests needed; existing tracing tests (`tracing.spec.ts`, `worker-tracing.spec.ts`) continue to pass
- [x] Breaking change?
  - No

---

# AI-assisted code disclosure (required)

## Was an AI tool used? (select one)
- [ ] No
- [x] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [ ] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot
- AI-assisted areas/files: `packages/durabletask-js/src/tracing/trace-helper.ts` — comment rewrite and dead code removal
- What you changed after AI output: Reviewed the analysis of OTEL proxy tracer behavior and validated the change is correct

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing

## Automated tests
- Result: Passed — 27 test suites, 464 tests (24 suites / 400 tests in durabletask-js, 3 suites / 64 tests in durabletask-js-azuremanaged)

## Manual validation (only if runtime/behavior changed)
- N/A — no runtime behavior change; this is a dead code removal and comment fix

---

# Notes for reviewers
- The OTEL JS SDK's `trace.getTracer()` returns a [proxy tracer](https://github.com/open-telemetry/opentelemetry-js/blob/main/api/src/trace/ProxyTracer.ts) that delegates to the current global provider at call time. This means the cached tracer automatically picks up provider changes without invalidation, making the previous `_cachedOtel` check unnecessary.

### CODE REVIEW

File: packages/durabletask-js/src/tracing/trace-helper.ts [hunk 1]
```diff
@@ -50,29 +69,27 @@ export interface OrchestrationSpanInfo {
...
   const spanName = createSpanName(TaskType.CREATE_ORCHESTRATION, name, version);
 
-  const span = tracer.startSpan(spanName, {
-    kind: otel.SpanKind.PRODUCER,
+  const span = ctx.tracer.startSpan(spanName, {
+    kind: ctx.otel.SpanKind.PRODUCER,
     attributes: {
-      [DurableTaskAttributes.TYPE]: TaskType.CREATE_ORCHESTRATION,
+      [DurableTaskAttributes.TYPE]: TaskType.ORCHESTRATION,
       [DurableTaskAttributes.TASK_NAME]: name,
```
[the remark this answers is not in the document]
Comment (torosent) [reply packages/durabletask-js/src/tracing/trace-helper.ts]: This is incorrect

File: packages/durabletask-js/test/tracing.spec.ts [hunk 2]
```diff
@@ -849,12 +850,120 @@ describe("Trace Helper - startSpanForEventRaisedFromClient", () => {
...
+
+    const result = createPbTraceContextFromSpan(nonRecordingSpan);
+    expect(result).toBeUndefined();
+  });
+
+  it("should correctly format traceparent with hex-padded flags (flags=0)", () => {
+    const tracer = otel.trace.getTracer(TRACER_NAME);
+    // Start a span; the default traceFlags from the SDK is SAMPLED (0x01),
+    // but we verify the hex padding by checking the formatted string.
+    const span = tracer.startSpan("pb-from-span-flags-test");
```
[the remark this answers is not in the document]
Comment (torosent) [reply packages/durabletask-js/test/tracing.spec.ts]: This is incorrect

**Review by YunchuWang — APPROVED** [review]
